src/components/actions/evaluatorFunctions.ts

```diff
@@ -8,6 +8,42 @@ Any changes done here should also be replicated in front-end
 
 import { DateTime, Duration } from 'luxon'
 
+type FilterRule = 'exclude' | 'include'
+interface FilterOptions {
+  key?: string
+  rule?: FilterRule
+  values: string | string[]
+}
+
+const filterArray = (valuesArray: Object[] | string[], options: FilterOptions) => {
+  const { key, rule = 'exclude', values } = options
+  if (
+    (valuesArray.length > 0 && key && typeof valuesArray[0] === 'object') ||
```

Comment:
`valuesArray.length > 0`

No need for this restriction, is there?

Reply:
Probably no need to check the type of the first item.

I would just check *each* item as part of the filter and if it's not an object then:
- exclude it if the rule is "include"
- include it if the rule is "exclude"

Then it follows the rule in a way that makes sense regardless of what type it is :)